test/classes.coffee

```diff
@@ -1808,3 +1808,19 @@ test 'Bound method of immediately instantiated class with expression base class
 
   {derivedBound} = a
   eq derivedBound(), 3
+
+test "#4591: super.x.y, super['x'].y", ->
```

Comment:
@vendethiel that changed in `2` to support ES-style `super` invocations. Compare [bare super], [super with parens], [super with access], and [super with both].

Deeper accesses against a `super` field could be useful if you're trying to `bind`/`call`/`apply` it, I guess.

```coffee
class A
  f: -> super.f.apply(this, arguments)
```

[bare super]: http://coffeescript.org/v2/#try:class%20A%0A%20%20f%3A%20-%3E%20super
[super with parens]: http://coffeescript.org/v2/#try:class%20A%0A%20%20f%3A%20-%3E%20super()
[super with access]: http://coffeescript.org/v2/#try:class%20A%0A%20%20f%3A%20-%3E%20super.f
[super with both]: http://coffeescript.org/v2/#try:class%20A%0A%20%20f%3A%20-%3E%20super.f()